apps/accounts/views.py

```diff
@@ -45,3 +45,25 @@ def get_auth_token(request):
 
     response_data = {"token": "{}".format(token)}
     return Response(response_data, status=status.HTTP_200_OK)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication, ))
+def new_auth_token(request):
+    try:
+        user = User.objects.get(email=request.user.email)
+    except User.DoesNotExist:
+        response_data = {"error": "This User account doesn't exist."}
+        return Response(response_data, status.HTTP_404_NOT_FOUND)
+    oldtoken = Token.objects.get(user=user)
+    if oldtoken:
+        Token.objects.filter(user=user).delete()
```

Comment:
There is no need to query again as you can delete the object you have obtained from the earlier query.